### PR TITLE
Add CommonSettings and ServiceApiSettings

### PR DESCRIPTION
**Pull request by chrisdunelm**
With tests.

### CODE REVIEW

File: src/Google.Api.Gax/Api/Gax/CommonSettings.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Common settings for all services.
+    /// </summary>
+    public abstract class CommonSettings
```
Comment (jskeet) [inline src/Google.Api.Gax/Api/Gax/CommonSettings.cs]: Possibly rename to ServiceSettingsBase?

File: src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Settings specifying the service API settings; how a client should connect to the service.
+    /// </summary>
+    public sealed class ServiceApiSettings
```
Comment (jskeet) [inline src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs]: ServiceEndpoint?

File: src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        private string _host;
+        private int? _port;
+
```
Comment (jskeet) [inline src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs]: Possibly have a constructor taking the host and port?

File: src/Google.Api.Gax/Api/Gax/CommonSettings.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
```
Comment (jskeet) [inline src/Google.Api.Gax/Api/Gax/CommonSettings.cs]: Missing copyright header on all files.

File: test/Google.Api.Gax.Tests/CommonSettingsTest.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public void Timeout_Invalid(int? millis)
+        {
+            var timeout = millis == null ? default(TimeSpan?) : TimeSpan.FromMilliseconds(millis.Value);
+            Assert.Throws<ArgumentOutOfRangeException>(() => new TestSettings { Timeout = timeout });
+        }
+
```
Comment (jskeet) [inline test/Google.Api.Gax.Tests/CommonSettingsTest.cs]: Cloning test, possibly with TestSettings overriding it in the expected way (as much for documentation as anything)?

File: src/Google.Api.Gax/Api/Gax/CommonSettings.cs (hunk 1 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax/Api/Gax/CommonSettings.cs]: Done

File: src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs (hunk 2 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax/Api/Gax/ServiceApiSettings.cs]: Done

File: src/Google.Api.Gax/Api/Gax/CommonSettings.cs (hunk 4 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax/Api/Gax/CommonSettings.cs]: Done

**Comment by jskeet** [comment]
LGTM.

**Pull request merged**